copier/copier.go

```diff
@@ -119,6 +63,66 @@ func extendedGlob(pattern string) (matches []string, err error) {
 	return matches, nil
 }
 
+// expandPatterns fans out a glob pattern so that "**" is expanded to all possible subdirectory paths at that location.
+func expandPatterns(pattern string) []string {
+	components := []string{}
+	dir := pattern
+	file := ""
+	for dir != filepath.VolumeName(dir) && dir != string(os.PathSeparator) {
+		dir, file = filepath.Split(dir)
+		if file != "" {
+			components = append([]string{file}, components...)
+		}
+		dir = strings.TrimSuffix(dir, string(os.PathSeparator))
+	}
+	patterns := []string{filepath.VolumeName(dir) + string(os.PathSeparator)}
+	for i := range components {
+		var nextPatterns []string
+		if components[i] == "**" {
+			for _, parent := range patterns {
+				nextSubdirs := subDirs(parent)
+				for _, nextSubdir := range nextSubdirs {
+					nextPatterns = append(nextPatterns, filepath.Join(parent, nextSubdir))
+				}
+			}
+		} else {
+			for _, parent := range patterns {
+				nextPattern := filepath.Join(parent, components[i])
+				nextPatterns = append(nextPatterns, nextPattern)
+			}
+		}
+		patterns = []string{}
+		seen := map[string]struct{}{}
+		for _, nextPattern := range nextPatterns {
+			if _, seen := seen[nextPattern]; seen {
+				continue
+			}
+			patterns = append(patterns, nextPattern)
+			seen[nextPattern] = struct{}{}
+		}
+	}
+	return patterns
+}
+
+// subDirs returns a recursive search of all directories under the given dir
```

Comment:
Style comment only, not necessary to change.  I'd rather see this function precede expandPatterns as expandPatterns calls it.